modules/apps/portal-url-builder/portal-url-builder-api/src/main/java/com/liferay/portal/url/builder/AbsolutePortalURLBuilder.java

```diff
@@ -36,6 +36,19 @@
 @ProviderType
 public interface AbsolutePortalURLBuilder {
 
+	/**
+	 * Returns a URL builder for ES modules. ES modules can usually be found
+	 * inside an OSGi bundle context path, in a subfolder of the '__liferay__'
+	 * folder.
+	 *
+	 * @param  webContextPath
+	 * the web context path containing the ES module (begins with /)
+	 * @param  modulePath the path to the ESM file (begins with /)
+	 * @return a URL builder for ES modules
+	 */
+	public ESModuleAbsolutePortalURLBuilder forESModule(
+		String webContextPath, String modulePath);
```

Comment:
sort?

maybe I am wrong about these sorting though, so don't take me too seriously

Reply:
The official position is not sort parameters alphabetically (though there are some exceptions) but only for new things.

Since this is a public API I don't think I should break it just for that. Also, since the goal of this PR is to make ESM work, I don't think we should add more noise than what's strictly necessary.

But yes, one thing is what I think, another what the PR review process deserves for this PR... :roll_eyes: